nim/app.py

```diff
@@ -34,13 +39,19 @@ def update_game_state():
 
 @app.route('/new', methods=['GET', 'POST'])
 def new_game():
-    json = loads('{"min": 3, "max": 12, "piles": 3}')
-    state = app.game.new_game(
-        min=json['min'],
-        max=json['max'],
-        piles=json['piles'],
-    )
+    """
+    route for /new so user can start new game
+    Uses JSON from frontend for params in new game
+    """
+    session['default_prefs'] = DEFAULT_GAME
+    user_request = request.get_json()
+    game_request = {**session.get('default_prefs', {}), **user_request}
+    try:
+        state = app.game.new_game(**game_request)
+    except NimException as e:
+        return jsonify({'error': e.args}), 400
     session['state'] = state
+    session['default_prefs'] = state
```

Comment:
This should be `session['default_prefs'] = game_request`, right?

Reply:
Yes, you're totally right. Changing now

Reply:
This should be `session['game_prefs']`. It's not a default if it keeps changing and being reset.